### PR TITLE
feat(query): support SELECT output nothing.

### LINKED ISSUE #8379: Feature: a SQL can output nothing.
**Summary**

Sometime, we just want to test the performance of a SQL and don't care about the output table. 
Like `do_something > /dev/null`.

We can introduce such a syntax:

```sql
SELECT * FROM t FORMAT NULL;
```

The result will be like:

```
mysql> select 1;
+------+
| 1    |
+------+
|    1 |
+------+
1 row in set (0.02 sec)
Read 1 rows, 1.00 B in 0.002 sec., 592.23 rows/sec., 592.23 B/sec.

mysql> select 1 FORMAT NULL;
1 row in set (0.02 sec)
Read 1 rows, 1.00 B in 0.002 sec., 592.23 rows/sec., 592.23 B/sec.
```

In the future, we can also add other output format .

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

```
mysql> select number from numbers(2);
+--------+
| number |
+--------+
|      0 |
|      1 |
+--------+
2 rows in set (0.04 sec)
Read 2 rows, 16.00 B in 0.004 sec., 445.89 rows/sec., 3.48 KiB/sec.

mysql> select number from numbers(2) ignore_result;
Empty set (0.03 sec)
Read 2 rows, 16.00 B in 0.004 sec., 462.98 rows/sec., 3.62 KiB/sec.
```

Fixes #8379

### CODE REVIEW

File: docs/doc/30-reference/30-sql/20-query-syntax/dml-select.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ SELECT
     [ORDER BY {col_name | expr | col_alias | col_position} [ASC | DESC], [ NULLS { FIRST | LAST }]
     [LIMIT row_count]
     [OFFSET row_count]
+    [FORMAT format]
```
Comment (youngsofun) [inline docs/doc/30-reference/30-sql/20-query-syntax/dml-select.md:23]: I prefer `IGNORE_RESULT` or `NO_OUTPUT` instead of reuse `FORMAT`

Comment (b41sh) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-select.md:23]: +1, `FORMAT NULL` may be confused.

Comment (sundy-li) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-select.md:23]: What's the problem of reusing the Null as a `format` name?

Comment (RinChanNOWWW) [reply docs/doc/30-reference/30-sql/20-query-syntax/dml-select.md]: > What's the problem of reusing the Null as a format name?

It's a little confused.

**Comment by sundy-li** [comment]
We should allow one of `FORMAT XXX | IGNORE_RESULT`

`select * from t format CSV IGNORE_RESULT` is not allowed.

**Comment by RinChanNOWWW** [comment]
> We should allow one of `FORMAT XXX | IGNORE_RESULT`
> 
> `select * from t format CSV IGNORE_RESULT` is not allowed.

`FORMAT` is only used in clickhouse handler and `IGNORE_RESULT` is only used in mysql handler.

**Comment by RinChanNOWWW** [comment]
@mergify update

**Review by sundy-li — APPROVED** [review]

**Comment by Xuanwo** [comment]
`select * from abc ignore_result` is not a good extension to me.

How about introducing a new statement like `PIPE INTO NULL select * from abc`?

In the future, we can extend it to `pipe into 's3://result' format csv select * from abc`.

**Comment by Xuanwo** [comment]
> How about introducing a new statement like `PIPE INTO NULL select * from abc`?

Result cache seems a particular case like `PIPE INTO @internal_stage/query-id.parquet select * from abc`. Maybe we can share some code in the future?

Cc @youngsofun for comments.

**Comment by RinChanNOWWW** [comment]
> PIPE INTO NULL select * from abc

It seems a nice idea. Is there any progress about this feature now?

**Comment by Xuanwo** [comment]
> It seems a nice idea. Is there any progress about this feature now?

No, it's a brand-new idea that came up one hour ago.

**Review by BohuTANG — APPROVED** [review]

**Comment by Xuanwo** [comment]
> If this feature looks good, we can start from `pipe into null` first.

most interesting idea from @BohuTANG: `COPY INTO NULL FROM (select * from abc)`

**Pull request merged**